js/shaders/scatter-vertex.glsl

```diff
@@ -0,0 +1,214 @@
+uniform float stroke_width;
+uniform float marker_scale;
+
+varying vec4 fill_color;
+varying vec4 stroke_color;
+varying vec3 vertex_position;
+varying vec2 vertex_uv;
+varying vec2 vUv;
+varying float pixel_size;
+
+// #ifdef AS_LINE
+// attribute vec3 position_previous;
+// #else
+attribute vec3 position;
+attribute vec2 uv;
+
+attribute float x;
+attribute float x_previous;
+attribute float y;
+attribute float y_previous;
+
+attribute float size;
+attribute float size_previous;
+
+attribute float rotation;
+attribute float rotation_previous;
+
+attribute float opacity;
+attribute float opacity_previous;
+
+// boolean not supported, so float will do
+attribute float selected;
+// #endif
+
+uniform sampler2D colormap;
+uniform sampler2D colormap_previous;
+attribute vec2 domain_color;
+
+#ifdef USE_COLORMAP
+attribute float color;
+#else
+attribute vec3 color;
+#endif
+
+#ifdef USE_COLORMAP_PREVIOUS
+attribute float color_previous;
+#else
+attribute vec3 color_previous;
+#endif
+
+
+
+#define SCALE_X(x) scale_transform_linear(x, range_x, domain_x)
+#define SCALE_Y(x) scale_transform_linear(x, range_y, domain_y)
+#define SCALE_SIZE(x) scale_transform_linear(x, range_size, domain_size)
+#define SCALE_ROTATION(x) scale_transform_linear(x, range_rotation, domain_rotation)
+#define SCALE_OPACITY(x) scale_transform_linear(x, range_opacity, domain_opacity)
+
+vec3 rotate_xy(vec3 x, float angle) {
+    float sina = sin(angle);
+    float cosa = cos(angle);
+    mat2 m = mat2(cosa, -sina, sina, cosa);
+    return vec3(m * x.xy, x.z);
+}
+
+void main(void) {
+
+    vec3 animation_time = vec3(animation_time_x, animation_time_y, animation_time_z);
+
+
+    vec3 domain_offset = vec3(domain_x.x, domain_y.x, domain_z.x);
+    vec3 domain_scale  = vec3(domain_x.y, domain_y.y, domain_z.y) - domain_offset;
+    vec3 range_offset = vec3(range_x.x, range_y.x, range_z.x);
+    vec3 range_scale  = vec3(range_x.y, range_y.y, range_z.y) - range_offset;
+
+    vec3 center = mix(vec3(x_previous, y_previous, 0), vec3(x, y, 0), animation_time);
+
+    // vec3 center_normalized = (center - domain_offset) / domain_scale;
+    // vec3 center_pixels = ((center_normalized*2.)-0.) * range_scale + range_offset;
+    // times 2 because the normalized coordinates go from [-1, 1]
+    vec3 center_pixels = vec3(SCALE_X(center.x), SCALE_Y(center.y), 0) * 2.;
+
+
+    // times 4 because of the normalized coordinates, and radius vs diameter use
+    pixel_size = sqrt(mix(SCALE_SIZE(size_previous), SCALE_SIZE(size), animation_time_size)) * marker_scale * 4.;
+    // we draw larger than the size for the stroke_width (on both side)
+    float s = pixel_size + 2.0 * stroke_width;
+    vUv = uv;
+    float angle = SCALE_ROTATION(mix(rotation_previous, rotation, animation_time_rotation));
+    vec3 model_pos = rotate_xy(position, 1.) * s + center_pixels;
+
+#ifdef USE_COLORMAP
+    float color_index = color; //(color - domain_color[0]) / (domain_color[1] - domain_color[0]);
```

Comment:
The `domain_color` value is never used and never updated, which results in the colormap use case not working. I'll see if I can fix it.

Reply:
It is defined as an `attribute` in shaders, but I guess it should be a `uniform`

Reply:
Fixed